### PR TITLE
[BouffaloLab]update sdk to support open network 

### PR DESCRIPTION
**Pull request by shchen-Lab**
[BouffaloLab]update sdk to support open network

### CODE REVIEW

File: src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp [hunk 1]
```diff
@@ -160,8 +160,14 @@ CHIP_ERROR BLWiFiDriver::ConnectWiFiNetwork(const char * ssid, uint8_t ssidLen,
     memcpy(passwd, key, keyLen);
     wifi_interface_t wifi_interface;
     wifi_interface = wifi_mgmr_sta_enable();
-    wifi_mgmr_sta_connect(&wifi_interface, wifi_ssid, passwd, NULL, NULL, 0, 0);
-
+    if (keyLen < 8 && keyLen != 5)
```
Comment (andy31415) [inline src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp:163]: Could you add a comment explaining this?
Why do we omit the password if the keyLength is less than 8? What is special about keyLen 5?

Comment (shchen-Lab) [reply src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp:163]: 1092b3c7991250f13c1a1b57efb2f70

 If keyLen is not equal 5 when keylength less 8 bytes， this is opennetwork.

Comment (andy31415) [reply src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp:163]: You did not add a comment. you should help readers of the code understand the code when reaching it.

A comment should exist (in code!) to explain this. Even better, 5 should be a named constant like `kWeb64PassPhraseLength`

Comment (andy31415) [reply src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp:163]: This also seems to indicate valid/invalid lengths. Why are we assuming 1,2,3,4,6,7 all being open? Should this not return an error for invalid length? Why is 8 a magic number from there (given that the list above only says 10 and 13 are valid).

Comment (shchen-Lab) [reply src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp:163]: What do you recommend return value ? like CHIP_ERROR_INVALID_STRING_LENGTH？

Comment (shchen-Lab) [reply src/platform/bouffalolab/BL602/NetworkCommissioningDriver.cpp]: I have update this at commit [aea5246], please help review this， thanks.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**